frontend/test/card_info_util_test.dart

```diff
@@ -41,5 +41,55 @@ void main() {
       final pepper = const Base64Decoder().convert("MvMjEqa0ulFDAgACElMjWA==");
       expect(cardInfo.hash(pepper), 'zogEJOhnSSp//8qhym/DdorQYgL/763Kfq4slWduxMg=');
     });
+
+    test("should be stable for a Nuernberg Pass with startDay", () {
+      final cardInfo = CardInfo()
+        ..fullName = 'Max Mustermann'
+        ..expirationDay = 365 * 40 // Equals 14.600
+        ..extensions = (CardExtensions()
+          ..extensionRegion = (RegionExtension()..regionId = 93)
+          ..extensionBirthday = (BirthdayExtension()..birthday = -365 * 10)
+          ..extensionNuernbergPassNumber = (NuernbergPassNumberExtension()..passNumber = 99999999)
+          ..extensionStartDay = (StartDayExtension()..startDay = 365 * 2));
+      final pepper = const Base64Decoder().convert("MvMjEqa0ulFDAgACElMjWA==");
+      expect(cardInfo.hash(pepper), '1ChHiAvWygwu+bH2yOZOk1zdmwTDZ4mkvu079cyuLjE=');
+    });
+  });
+
+  group("isCardNotYetValid", () {
+    test("should return true if startDay is in the future", () {
+      final cardInfo = CardInfo()
+        ..fullName = 'Max Mustermann'
+        ..expirationDay = 365 * 40 // Equals 14.600
+        ..extensions = (CardExtensions()
+          ..extensionRegion = (RegionExtension()..regionId = 93)
+          ..extensionBirthday = (BirthdayExtension()..birthday = -365 * 10)
+          ..extensionNuernbergPassNumber = (NuernbergPassNumberExtension()..passNumber = 99999999)
+          ..extensionStartDay = (StartDayExtension()..startDay = 365 * 70));
+      expect(isCardNotYetValid(cardInfo), true);
+    });
```

Comment:
Maybe, you could use https://pub.dev/packages/clock to test how `isCardNotYetValid` behaves when the time is very close to the `startDay`.
I think, ultimately, we should properly solve the timezone logic in the app. For the expiration day, we did some weird tolerance stuff.
Maybe, this can also be done in a seperate PR / issue though. I believe your current logic is "good enough" for now.

Reply:
https://github.com/digitalfabrik/entitlementcard/issues/1096